### ISSUE TITLE
-a option displays 2 notifications on volume change

### ISSUE THREAD

**Issue by darkowic** [christophgysin/pasystray#75]
Hi!

At start - I love your project <3

I have following problem: I run pasystray with -a option. When changing volume it displays 2 notifications instead of one. I use following commend to change volume sing keyboard shortcut:
```
amixer -q -D pulse set Master 2%- unmute
```

Arch linux, xfce

**Comment by Woowoo678** [comment]
would xfce4-volumed happen to be running?

**Comment by darkowic** [comment]
No, its not. And one instance of pasystray is on.

```
$ pidof pasystray 
495
```

**Comment by darkowic** [comment]
![screenshot_2016-11-20_22-59-39](https://cloud.githubusercontent.com/assets/11510581/20466092/b3afdb4e-af6c-11e6-910e-f52aef90aea3.png)

**Comment by Woowoo678** [comment]
Could you give the output of `dbus-monitor "interface='org.freedesktop.Notifications'"` when reproducing this issue?

**Comment by darkowic** [comment]
```
$ dbus-monitor "interface='org.freedesktop.Notifications'"
signal time=1481500376.160280 sender=org.freedesktop.DBus -> destination=:1.96 serial=2 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameAcquired
   string ":1.96"
signal time=1481500376.160340 sender=org.freedesktop.DBus -> destination=:1.96 serial=4 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string ":1.96"
method call time=1481500379.036385 sender=:1.48 -> destination=:1.30 serial=41 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=Notify
   string "pasystray"
   uint32 0
   string "audio-volume-low"
   string "sink Built-in Audio Analog Stereo  30%"
   string ""
   array [
   ]
   array [
      dict entry(
         string "urgency"
         variant             byte 0
      )
      dict entry(
         string "value"
         variant             int32 30
      )
   ]
   int32 2000
method call time=1481500379.046384 sender=:1.48 -> destination=:1.30 serial=42 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=Notify
   string "pasystray"
   uint32 0
   string "audio-volume-low"
   string "sink FFT based equalizer on Built-in Audio Analog Stereo  30%"
   string ""
   array [
   ]
   array [
      dict entry(
         string "urgency"
         variant             byte 0
      )
      dict entry(
         string "value"
         variant             int32 30
      )
   ]
   int32 2000
signal time=1481500381.060549 sender=:1.30 -> destination=(null destination) serial=89 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=NotificationClosed
   uint32 23
   uint32 1
signal time=1481500381.062499 sender=:1.30 -> destination=(null destination) serial=90 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=NotificationClosed
   uint32 24
   uint32 1
```

The difference that I found
```
string "sink Built-in Audio Analog Stereo  30%"
```
and
```
string "sink FFT based equalizer on Built-in Audio Analog Stereo  30%"
```

Looks like I have 2 sinks? Ehh I dont know how to debug it. 
My sinks
```
$ pacmd list-sinks
3 sink(s) available.
    index: 0
	name: <alsa_output.pci-0000_01_00.1.hdmi-stereo>
	driver: <module-alsa-card.c>
	flags: HARDWARE DECIBEL_VOLUME LATENCY DYNAMIC_LATENCY
	state: SUSPENDED
	suspend cause: IDLE 
	priority: 9050
	volume: front-left: 58982 /  90% / -2.75 dB,   front-right: 58982 /  90% / -2.75 dB
	        balance 0.00
	base volume: 65536 / 100% / 0.00 dB
	volume steps: 65537
	muted: no
	current latency: 0.00 ms
	max request: 0 KiB
	max rewind: 0 KiB
	monitor source: 0
	sample spec: s16le 2ch 44100Hz
	channel map: front-left,front-right
	             Stereo
	used by: 0
	linked by: 0
	configured latency: 0.00 ms; range is 0.50 .. 1999.82 ms
	card: 0 <alsa_card.pci-0000_01_00.1>
	module: 6
	properties:
		alsa.resolution_bits = "16"
		device.api = "alsa"
		device.class = "sound"
		alsa.class = "generic"
		alsa.subclass = "generic-mix"
		alsa.name = "HDMI 0"
		alsa.id = "HDMI 0"
		alsa.subdevice = "0"
		alsa.subdevice_name = "subdevice #0"
		alsa.device = "3"
		alsa.card = "1"
		alsa.card_name = "HDA NVidia"
		alsa.long_card_name = "HDA NVidia at 0xdf080000 irq 17"
		alsa.driver_name = "snd_hda_intel"
		device.bus_path = "pci-0000:01:00.1"
		sysfs.path = "/devices/pci0000:00/0000:00:01.0/0000:01:00.1/sound/card1"
		device.bus = "pci"
		device.vendor.id = "10de"
		device.vendor.name = "NVIDIA Corporation"
		device.product.id = "10f0"
		device.product.name = "GP104 High Definition Audio Controller"
		device.string = "hdmi:1"
		device.buffering.buffer_size = "352768"
		device.buffering.fragment_size = "176384"
		device.access_mode = "mmap+timer"
		device.profile.name = "hdmi-stereo"
		device.profile.description = "Digital Stereo (HDMI)"
		device.description = "GP104 High Definition Audio Controller Digital Stereo (HDMI)"
		alsa.mixer_name = "Nvidia GPU 83 HDMI/DP"
		alsa.components = "HDA:10de0083,14623302,00100100"
		module-udev-detect.discovered = "1"
		device.icon_name = "audio-card-pci"
	ports:
		hdmi-output-0: HDMI / DisplayPort (priority 5900, latency offset 0 usec, available: no)
			properties:
				device.icon_name = "video-display"
	active port: <hdmi-output-0>
  * index: 1
	name: <alsa_output.pci-0000_00_1f.3.analog-stereo>
	driver: <module-alsa-card.c>
	flags: HARDWARE HW_MUTE_CTRL HW_VOLUME_CTRL DECIBEL_VOLUME LATENCY DYNAMIC_LATENCY
	state: RUNNING
	suspend cause: 
	priority: 9959
	volume: front-left: 20711 /  32% / -30.02 dB,   front-right: 20711 /  32% / -30.02 dB
	        balance 0.00
	base volume: 65536 / 100% / 0.00 dB
	volume steps: 65537
	muted: no
	current latency: 507.89 ms
	max request: 168 KiB
	max rewind: 344 KiB
	monitor source: 1
	sample spec: s16le 2ch 44100Hz
	channel map: front-left,front-right
	             Stereo
	used by: 1
	linked by: 2
	configured latency: 980.00 ms; range is 0.50 .. 2000.00 ms
	card: 1 <alsa_card.pci-0000_00_1f.3>
	module: 7
	properties:
		alsa.resolution_bits = "16"
		device.api = "alsa"
		device.class = "sound"
		alsa.class = "generic"
		alsa.subclass = "generic-mix"
		alsa.name = "ALC1150 Analog"
		alsa.id = "ALC1150 Analog"
		alsa.subdevice = "0"
		alsa.subdevice_name = "subdevice #0"
		alsa.device = "0"
		alsa.card = "0"
		alsa.card_name = "HDA Intel PCH"
		alsa.long_card_name = "HDA Intel PCH at 0xdf240000 irq 322"
		alsa.driver_name = "snd_hda_intel"
		device.bus_path = "pci-0000:00:1f.3"
		sysfs.path = "/devices/pci0000:00/0000:00:1f.3/sound/card0"
		device.bus = "pci"
		device.vendor.id = "8086"
		device.vendor.name = "Intel Corporation"
		device.product.id = "a170"
		device.product.name = "Sunrise Point-H HD Audio"
		device.form_factor = "internal"
		device.string = "front:0"
		device.buffering.buffer_size = "352800"
		device.buffering.fragment_size = "176400"
		device.access_mode = "mmap+timer"
		device.profile.name = "analog-stereo"
		device.profile.description = "Analog Stereo"
		device.description = "Built-in Audio Analog Stereo"
		alsa.mixer_name = "Realtek ALC1150"
		alsa.components = "HDA:10ec0900,1462f984,00100001"
		module-udev-detect.discovered = "1"
		device.icon_name = "audio-card-pci"
	ports:
		analog-output-lineout: Line Out (priority 9900, latency offset 0 usec, available: yes)
			properties:
				
		analog-output-headphones: Headphones (priority 9000, latency offset 0 usec, available: no)
			properties:
				device.icon_name = "audio-headphones"
	active port: <analog-output-lineout>
    index: 2
	name: <equalizer>
	driver: <module-equalizer-sink.c>
	flags: HW_MUTE_CTRL LATENCY DYNAMIC_LATENCY
	state: SUSPENDED
	suspend cause: IDLE 
	priority: 1000
	volume: front-left: 20711 /  32%,   front-right: 20711 /  32%
	        balance 0.00
	base volume: 65536 / 100%
	volume steps: 65537
	muted: no
	current latency: 0.00 ms
	max request: 375 KiB
	max rewind: 689 KiB
	monitor source: 2
	sample spec: float32le 2ch 44100Hz
	channel map: front-left,front-right
	             Stereo
	used by: 0
	linked by: 0
	configured latency: 0.00 ms; range is 0.50 .. 2000.00 ms
	module: 26
	properties:
		device.description = "FFT based equalizer on Built-in Audio Analog Stereo"
		device.master_device = "alsa_output.pci-0000_00_1f.3.analog-stereo"
		device.class = "filter"
		device.icon_name = "audio-card"
```

**Comment by Woowoo678** [comment]
Ahh, so I was able to reproduce your issue by enabling an equalizer, myself. I guess since the equalizer creates a separate sink for itself, tied to the primary audio sink, pasystray is notifying for the changes to both of these sinks?

I guess that behavior _is_ technically correct, considering `-a` is meant for notifying _all_ pulseaudio changes, but awfully inconvenient no less. For what purpose are you using the `-a` option?

**Comment by darkowic** [comment]
I run pasystray in autostart with option -a. This is for support my keyboard's volume buttons.
When I run it without -a I doesn't show volume changes :/
![screenshot_2016-12-12_09-19-07](https://cloud.githubusercontent.com/assets/11510581/21091010/c6cce2ca-c043-11e6-9173-b78caddb8aa5.png)

**Comment by Woowoo678** [comment]
Until something is implemented here to better fit your need, you could try the "granular-notification-options" branch on my fork if you'd like. I'm not much of a C programmer or anything and it's a major work in progress, but your use case should be available through `pasystray --notify=sink_default`

**Comment by darkowic** [comment]
It means I have to compile it on my computer and install it hmm I've never done it so it can take a while... I don't have time for it now :/

**Comment by Woowoo678** [comment]
In that case, you could try the binary that I've compiled here: https://drive.google.com/open?id=0B58TImMMN8plcGgxVDJWdU9Rc3c

**Comment by christophgysin** [comment]
IIUC, this should be fixed with #82.
